File: Objective-C/AlertController_ObjC/Utility/Utility.m
//
//  Utility.m
//  AlertController_ObjC
//
//  Created by Hardik Trivedi on 06/11/19.
//  Copyright © 2019 9Brainz. All rights reserved.
//

#import "Utility.h"

@implementation Utility

//
//    // INPUT Parameter
//
//    // vc       : ViewController (in which you want to display Alert)
//    // title    : Title text of AlertController
//    // message  : Message text of AlertController
//    // button   : Multiple buttons for AlertController
//

+ (void)
shwoActionSheetInViewController: (UIViewController *)vc withTitle: (nullable NSString *)title andMessage: (nullable NSString *)message withButtons: (NSArray *)button cancelAvailable: (BOOL)isCancel withCompletion: (completion)completionBlock
{
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleActionSheet];
    
    for (int index = 0; index < button.count; index++) {
        
        UIAlertAction *action = [UIAlertAction actionWithTitle:button[index] style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
            
            if (completionBlock != nil) {

                completionBlock(index);
            }
        }];
        
        [alertController addAction:action];
    }
    
    if (isCancel) {
        
        UIAlertAction *action = [UIAlertAction actionWithTitle:[NSString stringWithFormat:@"Cancel"] style:UIAlertActionStyleDestructive handler:^(UIAlertAction * _Nonnull action) {
            
            if (completionBlock != nil) {

                completionBlock((int)[button count]);
            }
        }];
        
        [alertController addAction:action];
    }
    
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone) {

        // if device is iPhone
        [vc presentViewController:alertController animated:YES completion:nil];
    } else {
        
        // if device is iPad
        [alertController setModalPresentationStyle:UIModalPresentationPopover];
        
        UIPopoverPresentationController *popoverController = [alertController popoverPresentationController];
        [popoverController setSourceView:[vc view]];
        [popoverController setSourceRect:CGRectMake(CGRectGetMidX([[vc view] bounds]), CGRectGetMidY([[vc view] bounds]), 0, 0)];
        [popoverController setPermittedArrowDirections:UIPopoverArrowDirectionUnknown];
        
        [vc presentViewController:alertController animated:YES completion:nil];
    }
}

@end

File: Swift/ActionSheet_Swift/Utility/Utility.swift
//
//  Utility.swift
//  ActionSheet_Swift
//
//  Created by Hardik Trivedi on 07/11/19.
//  Copyright © 2019 9Brainz. All rights reserved.
//

import UIKit

class Utility: NSObject
{
    
    // INPUT Parameter
    
    // vc       : ViewController (in which you want to display Alert)
    // title    : Title text of AlertController
    // message  : Message text of AlertController
    // button   : Multiple buttons for AlertController
    
    static func shwoActionSheetInViewController(vc: UIViewController, withTitle title: String?, andMessage message: String?, withButtons button: [String], cancelAvailable isCancel: Bool, completion:((_ index: Int) -> Void)!) -> Void
    {
        let alertController = UIAlertController(title: title, message: message, preferredStyle: .actionSheet)

        for index in 0 ..< button.count  {

            let action = UIAlertAction(title: button[index], style: .default, handler: { (alert: UIAlertAction!) in

                if completion != nil {

                    completion(index)
                }
            })

            alertController.addAction(action)
        }
        
        if isCancel {
            
            let action = UIAlertAction(title: "Cancel", style: .destructive, handler: { (alert: UIAlertAction!) in

                if completion != nil {

                    completion(button.count)
                }
            })

            alertController.addAction(action)
        }
        
        if (UIDevice.current.userInterfaceIdiom == UIUserInterfaceIdiom.phone) {

            // if device is iPhone
            vc.present(alertController, animated: true, completion: nil)
        } else {
            
            // if device is iPad
            alertController.modalPresentationStyle = .popover

            if let popoverController = alertController.popoverPresentationController {
                
                popoverController.sourceView = vc.view
                popoverController.sourceRect = CGRect(x: vc.view.bounds.midX, y: vc.view.bounds.midY, width: 0, height: 0)
                popoverController.permittedArrowDirections = []
            }
        }
    }

}
